internal/corazawaf/rulegroup.go

```diff
@@ -133,6 +133,16 @@ func (rg *RuleGroup) Eval(phase types.RulePhase, tx *Transaction) bool {
 RulesLoop:
 	for i := range rg.rules {
 		r := &rg.rules[i]
+		// Check if a specific rule filter is applied to this transaction
+		// and if the current rule should be ignored according to the filter.
+		if tx.ruleFilter != nil {
```

Comment:
if you handle the nil in the struct itself I think you don't need to do this check.